perfschema/common.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package perfschema
+
+import (
+	"strings"
+
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/mysql"
+	"github.com/pingcap/tidb/util/codec"
+	"github.com/pingcap/tidb/util/types"
+)
+
+// castValues casts values based on columns type.
+func castValues(rec []interface{}, cols []*model.ColumnInfo) (err error) {
+	for _, c := range cols {
+		rec[c.Offset], err = types.Convert(rec[c.Offset], &c.FieldType)
+		if err != nil {
+			return errors.Trace(err)
+		}
+	}
+	return nil
+}
+
+// checkNotNull checks if nil value set to a column with NotNull flag is set.
+func checkNotNull(c *model.ColumnInfo, data interface{}) error {
+	if mysql.HasNotNullFlag(c.Flag) && data == nil {
+		return errors.Errorf("Column %s can't be null.", c.Name)
+	}
+	return nil
+}
+
+// checkNotNulls checks if row has nil value set to a set of columns with NotNull flag set.
+func checkNotNulls(cols []*model.ColumnInfo, row []interface{}) error {
```

Comment:
I am not sure if this is a problem, but I suggest we can revise it in later PRs if you really dislike this decision, is that ok?